core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -0,0 +1,458 @@
+    val oldFile = new File(sslProperties1.getProperty(SSL_KEYSTORE_LOCATION_CONFIG))
+    val newFile = File.createTempFile("keystore", ".jks")
+    Files.copy(oldFile.toPath, newFile.toPath, StandardCopyOption.REPLACE_EXISTING)
+    sslPropertiesCopy.setProperty(SSL_KEYSTORE_LOCATION_CONFIG, newFile.getPath)
+    alterSslKeystore(adminClient, sslPropertiesCopy, SecureInternal)
+    verifyProduceConsume(producer, consumer, 10, topic2)
+
+    // Verify that all messages sent with retries=0 while keystores were being altered were consumed
+    stopAndVerifyProduceConsume(producerThread, consumerThread, mayFailRequests = false)
+  }
+
+  private def createProducer(trustStore: File, retries: Int,
+                             clientId: String = "test-producer"): KafkaProducer[Array[Byte], Array[Byte]] = {
+    val bootstrapServers = TestUtils.bootstrapServers(servers, new ListenerName(SecureExternal))
+    val propsOverride = new Properties
+    propsOverride.put(ProducerConfig.CLIENT_ID_CONFIG, clientId)
+    val producer = TestUtils.createNewProducer(bootstrapServers, acks = -1, retries = retries,
+      securityProtocol = SecurityProtocol.SASL_SSL, trustStoreFile = Some(trustStore),
+      saslProperties = Some(clientSaslProps), props = Some(propsOverride))
+    producers += producer
+    producer
+  }
+
+  private def createConsumer(groupId: String, trustStore: File, topic: String = topic):KafkaConsumer[Array[Byte], Array[Byte]] = {
+    val bootstrapServers = TestUtils.bootstrapServers(servers, new ListenerName(SecureExternal))
+    val consumer = TestUtils.createNewConsumer(bootstrapServers, groupId,
+      securityProtocol = SecurityProtocol.SASL_SSL, trustStoreFile = Some(trustStore),
+      saslProperties = Some(clientSaslProps))
+    consumer.subscribe(Collections.singleton(topic))
+    consumers += consumer
+    consumer
+  }
+
+  private def createAdminClient(securityProtocol: SecurityProtocol, listenerName: String): AdminClient = {
+    val config = new util.HashMap[String, Object]
+    val bootstrapServers = TestUtils.bootstrapServers(servers, new ListenerName(listenerName))
+    config.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers)
+    val securityProps: util.Map[Object, Object] =
+      TestUtils.adminClientSecurityConfigs(securityProtocol, Some(trustStoreFile1), Some(clientSaslProps))
+    securityProps.asScala.foreach { case (key, value) => config.put(key.asInstanceOf[String], value) }
+    val adminClient = AdminClient.create(config)
+    adminClients += adminClient
+    adminClient
+  }
+
+  private def verifyProduceConsume(producer: KafkaProducer[Array[Byte], Array[Byte]],
+                                   consumer: KafkaConsumer[Array[Byte], Array[Byte]],
+                                   numRecords: Int,
+                                   topic: String = topic): Unit = {
+    val producerRecords = (1 to numRecords).map(i => new ProducerRecord(topic, s"key$i".getBytes,
```

Comment:
Not a big deal, but if we want to use strings, we can use their corresponding serializers.